app/stac_api/models.py

```diff
@@ -125,48 +145,137 @@ class Collection(models.Model):
+                self.save()
+
+            elif asset_proj_epsg not in self.summaries["proj:epsg"]:
+                self.summaries["proj:epsg"].append(asset_proj_epsg)
+                self.save()
+
+            if self.summaries["eo:gsd"] is None:
+                self.summaries["eo:gsd"] = [asset_eo_gsd]
+                self.save()
+
+            elif not float_in(asset_eo_gsd, self.summaries["eo:gsd"]):
+                self.summaries["eo:gsd"].append(asset_eo_gsd)
+                self.save()
+
+        except (KeyError, IndexError) as err:
+            logger.error(_(
+                "Error when updating collection's summaries values due to asset update: %s"), err
+                        )
+            raise ValidationError(_(
+                "Error when updating collection's summaries values due to asset update."
+            ))
+
+    def update_extent(self, item_properties_datetime):
+        '''
+        updates the collection's temporal extent when item's are update.
+        :param item_properties_datetime: item's value for properties_datetime.
+        This function checks, if the corresponding parameter of the collection
+        needs to be updated. If so, it will be either updated or an error will
+        be raised, if updating fails.
+        '''
+        try:
+
+            if self.extent["temporal"]["interval"][0][0] is None:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif item_properties_datetime < self.extent["temporal"]["interval"][0][0]:
+
+                self.extent["temporal"]["interval"][0][0] = item_properties_datetime
+                self.save()
+
+            elif self.extent["temporal"]["interval"][0][1] is None:
+
+                self.extent["temporal"]["interval"][0][1] = item_properties_datetime
+                self.save()
+
+            elif item_properties_datetime > self.extent["temporal"]["interval"][0][1]:
+
+                self.extent["temporal"]["interval"][0][1] = item_properties_datetime
+                self.save()
+
+        except (KeyError, IndexError) as err:
+
+            logger.error(_('Updating the collection extent due to item update failed: %s'), err)
+            raise ValidationError(_("Updating the collection extent due to item update failed."))
+
+    def clean(self):
         # very simple validation, raises error when geoadmin_variant strings contain special
         # characters or umlaut.
-        for variant in self.geoadmin_variant:
-            if not bool(re.search('^[a-zA-Z0-9]*$', variant)):
-                raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+
+        try:
+            for variant in self.summaries["geoadmin:variant"]:
+                if not bool(re.search('^[a-zA-Z0-9]*$', variant)):
+                    raise ValidationError(_("Property geoadmin:variant not compatible with the" +
```

Comment:
A log message would be appropriate here as well.

Reply:
yeah, why not :-)